decidim-admin/app/helpers/decidim/admin/newsletters_helper.rb

```diff
@@ -101,24 +93,32 @@ def sent_to_scopes(newsletter)
       end
 
       def organization_participatory_space(manifest_name)
-        @organization_participatory_space ||= Decidim.find_participatory_space_manifest(manifest_name)
-                                                     .participatory_spaces.call(current_organization)&.published&.order(title: :asc)
+        @organization_participatory_spaces ||= {}
+        @organization_participatory_spaces[manifest_name] ||= Decidim.find_participatory_space_manifest(manifest_name)
+                                                                     .participatory_spaces.call(current_organization)&.published&.order(title: :asc)
       end
 
       def spaces_user_can_admin
```

Comment:
is this method returning `spaces_user_can_admin_as_options_for_select`?

Reply:
This method generates a `Hash` of the current_organizations published participatory_spaces, being the `manifest_name` the key and an `ActiveRecord::Relation` the value, such as:

```rb
{
  assemblies: ActiveRecord::Relation[
    <Decidim::Assembly id: 2 ... >, 
    <Decidim::Assembly id: 3 ... >
  ],
  conferences: ActiveRecord::Relation [
    <Decidim::Conference id: 1 ... >,
    <Decidim::Conference id: 2 ... >
  ],
  consultations: ActiveRecord::Relation [
    <Decidim::Consultation id: 1 .. >
  ], 
  initiatives: ActiveRecord::Relation [
    <Decidim::Initiative id: 6 ... >
  ], 
  participatory_processes: ActiveRecord::Relation [
    <Decidim::ParticipatoryProcess id: 2 ... >,
    <Decidim::ParticipatoryProcess id: 4 ... >
  ]
}
```
so when called `organization_participatory_space(:assemblies)`, returns an `ActiveRecord::Relation` of all published _asseblies_ of the organization.

Reply:
Returns the structure for the select, used in:

https://github.com/gencat/decidim/blob/c00248d6bca208bb214ca0851e0229556b1c5279/decidim-admin/app/helpers/decidim/admin/newsletters_helper.rb#L36-L41

**but it's also used to check** which spaces the user can manage:

https://github.com/gencat/decidim/blob/c00248d6bca208bb214ca0851e0229556b1c5279/decidim-admin/app/helpers/decidim/admin/newsletters_helper.rb#L15-L16